test/fn/move-contacts.spec.js

```diff
@@ -0,0 +1,265 @@
+      return undefined;
+    }
+
+    // Remove _rev because it makes expectations harder to write
+    delete result._rev;
+    return result;
+  };
+
+  for (let webappVersion of [2, 3]) {
+    describe(`webapp views for v${webappVersion}.x`, () => {
+      beforeEach(async () => {
+        pouchDb = new PouchDB(`scenario${scenarioCount++}`, { adapter: 'memory' });
+
+        await mockHierarchy(pouchDb, {
+          district_1: {
+            health_center_1: {
+              clinic_1: {
+                patient_1: {},
+              },
+            },
+          },
+          district_2: {},
+        });
+
+        await pouchDb.put({ _id: 'settings', settings: {} });
+
+        await mockReport(pouchDb, {
+          id: 'report_1',
+          creatorId: 'health_center_1_contact',
+        });
+
+        const views = { reports_by_freetext };
+        if (webappVersion > 2) {
+          // contacts_by_place is is new in 3.0.x
+          views.contacts_by_place = contacts_by_place;
+        } else {
+          // 3.x defines contacts_by_parent but the key has changed format. Don't allow its use.
+          views.contacts_by_parent = contacts_by_parent;
+        }
+
+        await pouchDb.put({
+          _id: '_design/medic-client',
+          views,
+        });
+
+        moveContactsModule.__set__('writeDocumentToDisk', (docDirectoryPath, doc) => writtenDocs.push(doc));
+        writtenDocs.length = 0;
+      });
+
+      afterEach(async () => pouchDb.destroy());
+
+      it('move health_center_1 to district_2', async () => {
+        await updateLineagesAndStage({
+          contactIds: ['health_center_1'],
+          parentId: 'district_2',
+        }, pouchDb);
+
+        expect(await getWrittenDoc('health_center_1_contact')).to.deep.eq({
+          _id: 'health_center_1_contact',
+          type: 'person',
+          parent: parentsToLineage('health_center_1', 'district_2'),
+        });
+
+        expect(await getWrittenDoc('health_center_1')).to.deep.eq({
+          _id: 'health_center_1',
+          type: 'health_center',
+          contact: { _id: 'health_center_1_contact' },
+          parent: parentsToLineage('district_2'),
+        });
+
+        expect(await getWrittenDoc('clinic_1')).to.deep.eq({
+          _id: 'clinic_1',
+          type: 'clinic',
+          contact: { _id: 'clinic_1_contact' },
+          parent: parentsToLineage('health_center_1', 'district_2'),
+        });
+
+        expect(await getWrittenDoc('patient_1')).to.deep.eq({
+          _id: 'patient_1',
+          type: 'person',
+          parent: parentsToLineage('clinic_1', 'health_center_1', 'district_2'),
+        });
+
+        expect(await getWrittenDoc('report_1')).to.deep.eq({
+          _id: 'report_1',
+          form: 'foo',
+          type: 'data_record',
+          contact: parentsToLineage('health_center_1_contact', 'health_center_1', 'district_2'),
+        });
+      });
+
+      it('move health_center_1 to root', async () => {
```

Comment:
Note to self: 

* Add a test for setting contact's parent as to self
* Add a test for moving a contact to have the same parent and confirm no change